react-front-end/tsrc/modules/ViewerModule.ts

```diff
@@ -248,3 +263,184 @@ export const buildLightboxNavigationHandler = (
+              uuid,
+              version,
+              attachment,
+              viewerId?.right
+            ),
+          });
+    })
+  );
+  const lefts: string[] = A.lefts(either);
+  const rights: AttachmentAndViewerDefinition[] = A.rights(either);
+
+  return A.isEmpty(lefts) ? E.right(rights) : E.left(lefts.join());
+};
+
+/**
+ * Build Lightbox entries based on a list of attachments and viewer definitions
+ *
+ * @param attachmentsAndViewerDefinitions A list of attachments and their viewer definitions.
+ */
+export const buildLightboxEntries = (
+  attachmentsAndViewerDefinitions: AttachmentAndViewerDefinition[]
+): LightboxEntry[] =>
+  attachmentsAndViewerDefinitions
+    .filter(({ viewerDefinition: [viewer] }) => viewer === "lightbox")
+    .map(
+      ({
+        attachment: { id, description, mimeType },
+        viewerDefinition: [_, src],
+      }) => ({
+        src,
+        title: description,
+        mimeType: mimeType ?? "",
+        id,
+      })
+    );
+
+/**
+ * Convert attachments' viewer definitions to their viewer configurations.
+ *
+ * @param attachmentsAndViewerDefinitions A list of attachments and their viewer definitions.
+ * @param lightboxEntries A list of attachments that are viewable in the Lightbox.
+ */
+export const convertViewerDefinitionToViewerConfig = (
+  attachmentsAndViewerDefinitions: AttachmentAndViewerDefinition[],
+  lightboxEntries: LightboxEntry[]
+): AttachmentAndViewerConfig[] =>
+  attachmentsAndViewerDefinitions.map(
+    ({ viewerDefinition: [viewer, viewUrl], attachment }) => {
+      const initialLightboxEntryIndex = lightboxEntries.findIndex(
+        (entry) => entry.id === attachment.id
+      );
+      return viewer === "lightbox"
+        ? {
+            attachment,
+            viewerConfig: {
+              viewerType: viewer,
+              config: {
+                src: viewUrl,
+                title: attachment.description,
+                mimeType: attachment.mimeType ?? "",
+                onNext: buildLightboxNavigationHandler(
+                  lightboxEntries,
+                  initialLightboxEntryIndex + 1
+                ),
+                onPrevious: buildLightboxNavigationHandler(
+                  lightboxEntries,
+                  initialLightboxEntryIndex - 1
+                ),
+              },
+            },
+          }
+        : {
+            attachment,
+            viewerConfig: {
+              viewerType: viewer,
+              url: viewUrl,
+            },
+          };
+    }
+  );
+
+/**
+ * Returns a list of provided attachments and their viewer configurations.
+ *
+ * @param attachments Attachments provided to find their viewer configurations.
+ * @param uuid UUID of the Item which the attachments belong to.
+ * @param version Version of the Item which the attachments belong to.
+ * @param getViewerDetails Function called to retrieve viewer detail for each attachment.
+ */
+export const buildViewerConfigForAttachment = async (
```

Comment:
Shouldn't this be plural?

```suggestion
export const buildViewerConfigForAttachments = async (
```